easybuild/easyconfigs/l/louvain/louvain-0.8.0-foss-2021a.eb

```diff
@@ -0,0 +1,45 @@
+easyblock = 'PythonBundle'
+
+name = 'louvain'
+version = '0.8.0'
+
+homepage = 'https://pypi.org/project/louvain'
+description = "louvain is a general algorithm for methods of community detection in large networks"
+
+toolchain = {'name': 'foss', 'version': '2021a'}
+
+builddependencies = [
+    ('CMake', '3.20.1'),
+    ('flex', '2.6.4'),
+    ('Bison', '3.7.6'),
+]
+
+dependencies = [
+    ('Python', '3.9.5'),
+]
+
+use_pip = True
+
+# fix failing CMake command when building vendored igraph because of missing 'interfaces' subdirectory
+local_louvain_preinstallopts = "mkdir -p vendor/source/igraph/interfaces && "
+local_louvain_preinstallopts += "touch vendor/source/igraph/interfaces/CMakeLists.txt && "
+
+# trigger build of vendored igraph in vendor/source/igraph
+local_louvain_preinstallopts += "python setup.py build && "
```

Comment:
why build the vendored `igraph` at all? according to the docs:
```
 63 There are basically two installation modes, similar to the python-igraph package                                        
 64 itself (from which most of the setup.py comes).                                                                         
 65                                                                                                                         
 66 1. No ``C`` core library is installed yet. The ``C`` core                                                               
 67    library of igraph that is provided within the ``louvain-igraph`` package is                                          
 68    compiled.                                                                                                            
 69 2. A ``C`` core library is already installed. In this case, you may link                                                
 70    dynamically to the already installed version by specifying                                                           
 71    ``--no-pkg-config``. This is probably also the version that is used by the                                           
 72    igraph package, but you may want to double check this.                                                               
 73                                                                                                                         
 74 In case the ``python-igraph`` package is already installed before, make sure that                                       
 75 both use the **same versions** (at least the same minor version, which should be                                        
 76 API compatible).
```

Reply:
To minimize problems, it is best to link this package and python-igraph to the same igraph C library.